src/ReplicatedStorage/Client/InstanceComponents/Misc/FallDamage.lua

```diff
@@ -5,30 +5,20 @@ local Trove = require(ReplicatedStorage.Packages.Trove)
 local FallDamage = {}
 FallDamage.__index = FallDamage
 
-function FallDamage:_trackCharacterFall(character: Model, humanoid: Humanoid)
-	local initialHeight = nil
-	local fallHeight = nil
-
-	self._trove:Connect(humanoid.FreeFalling, function(active)
-		if active then
-			initialHeight = character.HumanoidRootPart.Position.Y
-		elseif initialHeight then
-			fallHeight = initialHeight - character.HumanoidRootPart.Position.Y
-			if fallHeight < FallDamageConfig.Threshold then return end
-
-			local damage = (fallHeight - FallDamageConfig.Threshold) * FallDamageConfig.Scale
-			humanoid:TakeDamage(damage)
-		end
-	end)
-end
-
 function FallDamage.new(character: Model)
 	local humanoid = character:WaitForChild("Humanoid")
 
 	local self = setmetatable({}, FallDamage)
 
 	self._trove = Trove.new()
-	self:_trackCharacterFall(character, humanoid)
+	self._trove:Connect(humanoid.StateChanged, function(old, new)
+		if not (old == Enum.HumanoidStateType.Freefall and new == Enum.HumanoidStateType.Landed) then return end
+		
+		local humanoidRootPart = character:WaitForChild("HumanoidRootPart")
```

Comment:
I'm pretty sure we should be accessing this via Humanoid.RootPart and not searching for HumanoidRootPart

Reply:
@Hexcede Your thoughts?

Reply:
It cannot because if the character model has died, it will index nil with humanoid root part.

Reply:
> It cannot because if the character model has died, it will index nil with humanoid root part.

`humanoid.RootPart` will be `nil` if the root part is destroyed. It avoids FindFirstChild, and it avoids WaitForChild. Problem is you can possibly leak some memory here for the thread. In any case, FindFirstChild is more appropriate in this connection not WaitForChild, and `humanoid.RootPart` would probably be better overall I bet.

Reply:
I believe it's also more correct to use .RootPart as it's avoiding a dependency on the name of the RootPart.